### ISSUE TITLE
Infer non-string types from property values.

### ISSUE THREAD

**Issue by skwirking** [FasterXML/jackson-dataformats-text#20]
It would be great if there was a way to specify a non-string property value when mapping to the generic `ObjectNode` type. For example:

```
Properties props = new Properties();

props.setProperty("booleanExample", "true");
props.setProperty("integerExample", "4");
props.setProperty("stringExample", "some string");
    
ObjectNode root = new JavaPropsMapper().readValue(props, ObjectNode.class);

System.out.println(new ObjectMapper().writeValueAsString(root));
```

This code outputs

`{"integerExample":"4","booleanExample":"true","stringExample":"some string"}`

But it would be great if there was some configuration or way to escape property values such that the output inferred integer or boolean types from the values themselves:

`{"integerExample":4,"booleanExample":true,"stringExample":"some string"}`

**Comment by cowtowncoder** [comment]
Yes, that would make sense. Not sure where and how this would work: I suspect that simple setting would not suffice since users tend to have their own preferences and nuances. So most likely need a handler of some kind.
Which leads to something like maybe:

https://github.com/FasterXML/jackson-core/issues/355

which would allow adding hooks for such conversion at low level: and could then tackle both properties and CSV format -- and perhaps even YAML, and maybe if we are lucky XML.

**Comment by skwirking** [comment]
Thanks, @cowtowncoder. I hadn't seen that issue. Perhaps this could be useful as a real-world use case for that feature once development starts.

**Comment by cowtowncoder** [comment]
@skwirking yes, absolutely.

**Comment by mschielmann** [comment]
Hi, a lot of time has passed - was just wondering - has anything possibly changed in regards to that issue, or is there any other way that JavaPropsMapper would read to JsonNode having integers / booleans instead of TextNodes?

**Comment by cowtowncoder** [comment]
@mschielmann No, nothing has changed. If anyone had time & interest in adding configurable coercion/detection functionality, I'd be happy to review.
I will not have time to work on this in near future but I make it my priority to help others.

**Comment by mschielmann** [comment]
Thank you for your answer @cowtowncoder, I of course understand.
Seems like an interesting item to pick up, but quite honestly, I feel a bit lost trying to understand how this would fit into the whole ecosystem.

**Comment by cowtowncoder** [comment]
@mschielmann Yes, it's not necessarily the simplest thing ever. But changes would be quite self-contained; in `JavaPropertiesParser` (or however it was named), before indicating something as `JsonToken.VALUE_STRING` check against a regexp for numbers (etc) to see if coercion would be needed. But yeah then state keeping will become more challenging... and in case of numbers, magnitude (to expose as `int`, `long` or `BigInteger` for integrals; `double` or `BigDecimal` for floating-point) will need non-trivial handling.

Similar questions woudl be relevant for CSV backend as well, fwtw.
